main.go

```diff
@@ -105,6 +110,11 @@ func headerKeyValue(h string) (string, string) {
 // visit visits a url and times the interaction.
 // If the response is a 30x, visit follows the redirect.
 func visit(url *url.URL) {
+	if redirectsFollowed > maxRedirects {
```

Comment:
Could you please move this logic down into the follow redirect code around line 257